lib/teleterm/daemon/gateway_cert_reissuer_test.go

```diff
@@ -0,0 +1,292 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package daemon
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	api "github.com/gravitational/teleport/lib/teleterm/api/protogen/golang/v1"
+	"github.com/gravitational/teleport/lib/teleterm/api/uri"
+	"github.com/gravitational/teleport/lib/teleterm/clusters"
+	"github.com/gravitational/teleport/lib/teleterm/gateway"
+	"github.com/gravitational/teleport/lib/tlsca"
+)
+
+var log = logrus.WithField(trace.Component, "reissuer")
+
+func TestReissueCert_CallsDBCertReissuerOnceIfItReturnsSuccessfully(t *testing.T) {
```

Comment:
We can run these tests in `t.Parallel()`, right?

Reply:
> We can run these tests in `t.Parallel()`, right?

Yes, but is it worth it? It's always hard for me to judge that. Those tests are rather lightweight, they don't perform any heavy IO operations. Is there a rule of thumb as to when we should use `t.Parallel` for tests?

Reply:
I personally treat it like if test is not the simplest/quickest one it already can be worth adding `t.Parallel` and let go figure it out, who knows, maybe tests will run on a 256-core arm etc. But yeah, I guess for these tests it doesn't matter much.